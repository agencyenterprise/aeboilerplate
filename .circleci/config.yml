version: 2
jobs:
  build:
    docker:
      - image: circleci/node:8.11.0
      - image: postgres:10.4
        environment:
          POSTGRES_USER: ubuntu
          POSTGRES_DB: api-db
    working_directory: ~/repo
    steps:
      - checkout
      - run:
          name: Waiting for Postgres to be ready
          command: |
            for i in `seq 1 10`;
            do
              nc -z localhost 5432 && echo Success && exit 0
              echo -n .
              sleep 1
            done
            echo Failed waiting for Postgress && exit 1
      - run:
          name: Install API packages
          command: |
            cd api
            npm i
      - run:
          name: Run API lint
          command: |
            cd api
            npm run lint
      - run:
          name: Run API tests
          environment:
            NODE_ENV: test
            DATABASE_URL: 'postgres://ubuntu@localhost:5432/api-db?sslmode=disable'
          command: |
            cd api
            set -ou pipefail
            npm run test-coverage
      - setup_remote_docker
      - run:
          name: Run generator
          environment:
            NODE_ENV: ci
          command: |
            if [ "${GENERATOR_ENABLED}" == "true" ]; then
              set -ou pipefail
              npm i
              npm run generator
            fi
      - restore_cache:
          keys:
            - v1-dependencies-{{ checksum "package.json" }}
            - v1-dependencies-
      - save_cache:
          paths:
            - node_modules
          key: v1-dependencies-{{ checksum "package.json" }}
      # - deploy:
      #     command: |
      #       if [ "${CIRCLE_BRANCH}" == "master" ]; then
      #         bash ./.circleci/deploy-heroku.sh
      #       fi
