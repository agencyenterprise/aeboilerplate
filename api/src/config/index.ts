export const config = {
  http: {
    host: process.env.HTTP_HOST || '0.0.0.0',
    port: process.env.VIRTUAL_PORT || process.env.PORT || 3000,
  },
  auth: {
    google: {
      clientID: process.env.GOOGLE_ID,
      clientSecret: process.env.GOOGLE_SECRET,
      callbackURL: process.env.GOOGLE_CALLBACK_URL,
      scope: ['https://www.googleapis.com/auth/userinfo.profile', , 'https://www.googleapis.com/auth/userinfo.email'],
    },
    facebook: {
      clientID: process.env.FACEBOOK_ID,
      clientSecret: process.env.FACEBOOK_SECRET,
      callbackURL: process.env.FACEBOOK_CALLBACK_URL,
      profileFields: ['id', 'name', 'short_name', 'photos', 'email'],
    },
    linkedIn: {
      clientID: process.env.LINKEDIN_ID,
      clientSecret: process.env.LINKEDIN_SECRET,
      callbackURL: process.env.LINKEDIN_CALLBACK_URL,
      scope: ['r_emailaddress', 'r_basicprofile'],
    },
    providers: {
      google: 'googleProvider',
      facebook: 'facebookProvider',
      linkedIn: 'linkedInProvider',
    },
    successLoginRedirectUrl: process.env.SUCCESS_LOGIN_REDIRECT_URL,
  },
  logLevel: process.env.LOG_LEVEL || 'debug',
  env: process.env.NODE_ENV || 'development',
  nodeClusterEnabled:
    (process.env.ENABLE_NODE_CLUSTER && process.env.ENABLE_NODE_CLUSTER.toLowerCase() === 'true') || false,
  knex: {
    client: 'postgresql',
    connection: process.env.DATABASE_URL || {
      port: process.env.DB_CONN_PORT || 5432,
      host: process.env.DB_CONN_HOST || 'localhost',
      database: process.env.DB_CONN_DATABASE || 'api-db',
      user: process.env.DB_CONN_USER || 'user',
      password: process.env.DB_CONN_PASSWORD || 'password',
      multipleStatements: true,
    },
    pool: {
      min: +process.env.DATABASE_POOL_MIN || 0,
      max: +process.env.DATABASE_POOL_MAX || 10,
    },
    migrations: {
      directory: `${__dirname}/../../db/pg/migrations`,
      tableName: 'knex_migrations',
    },
    seeds: {
      directory: `${__dirname}/../../db/pg/seeds`,
    },
  },
}
