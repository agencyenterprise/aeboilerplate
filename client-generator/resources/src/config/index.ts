export const config = {
  apiUrl: process.env.REACT_APP_API_URL,
  localStorageKeys: {
    token: 'token',
  },
  auth: {
    linkedInAuthURL: process.env.REACT_APP_LINKEDIN_AUTH_URL,
  },
}
