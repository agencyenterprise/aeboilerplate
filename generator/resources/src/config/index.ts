export const config = {
  apiUrl: process.env.REACT_APP_API_URL || '/api',
  localStorageKeys: {
    token: 'token',
  },
  authUrl: {
    google: process.env.REACT_APP_GOOGLE_AUTH_URL || '/api/auth/google',
    linkedIn: process.env.REACT_APP_LINKEDIN_AUTH_URL || '/api/auth/linkedin',
    facebook: process.env.REACT_APP_FACEBOOK_AUTH_URL || '/api/auth/facebook',
  },
}
