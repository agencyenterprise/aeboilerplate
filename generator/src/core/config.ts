export const clientAppPath = './client'
export const resourcesFilesPath = './generator/resources'
export const totalSteps = 10
