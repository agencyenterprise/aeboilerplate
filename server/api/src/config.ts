const configuration = {
  http: {
    host: process.env.HTTP_HOST || '0.0.0.0',
    port: process.env.PORT || 3000,
  },
  auth: {
    linkedIn: {
      clientID: '77gy1y1ddffvxw',
      clientSecret: 'WNOSE0fM0KdVPlDq',
      callbackURL: 'http://localhost:3000/auth/linkedin/callback',
      scope: ['r_emailaddress', 'r_basicprofile'],
    },
  },
  logLevel: process.env.LOG_LEVEL || 'debug',
  env: process.env.NODE_ENV || 'development',
}

export default configuration
