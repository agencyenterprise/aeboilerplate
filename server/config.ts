export const config = {
  http: {
    host: process.env.HTTP_HOST || '0.0.0.0',
    port: process.env.PORT || 3000,
  },
  env: process.env.NODE_ENV || 'development',
  nodeClusterEnabled: (process.env.ENABLE_NODE_CLUSTER && process.env.ENABLE_NODE_CLUSTER.toLowerCase() === 'true') || false,
}
