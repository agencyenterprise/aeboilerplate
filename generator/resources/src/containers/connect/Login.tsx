import * as React from 'react'

import { config } from '../../config'

import './login.scss'

const Login = () => (
  <div className="login">
    <p>Sign in with:</p>
    <a href={config.authUrl.google}>
      <button>Google</button>
    </a>
    <a href={config.authUrl.facebook}>
      <button>Facebook</button>
    </a>
    <a href={config.authUrl.linkedIn}>
      <button>LinkedIn</button>
    </a>
  </div>
)

export { Login }
