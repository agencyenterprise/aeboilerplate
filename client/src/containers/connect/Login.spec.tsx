import { configure, shallow } from 'enzyme'
import * as Adapter from 'enzyme-adapter-react-16'
import * as React from 'react'
import * as renderer from 'react-test-renderer'

import { config } from '../../config'
import { Login } from './Login'

configure({ adapter: new Adapter() })

describe(`<Login />`, () => {
  it('renders', () => {
    const login = renderer.create(<Login />).toJSON()
    expect(login).toMatchSnapshot()
  })

  it('has login class', () => {
    expect(shallow(<Login />).hasClass('login')).toBeTruthy()
  })

  it('has a link pointing to google path', () => {
    expect(
      shallow(<Login />)
        .find('a')
        .prop('href'),
    ).toEqual(config.auth.googleAuthURL)
  })
})
