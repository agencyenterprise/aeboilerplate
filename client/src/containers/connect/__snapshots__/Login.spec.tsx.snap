// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Login /> renders 1`] = `
<div
  className="login"
>
  <a
    href="/api/auth/google"
  >
    <button>
      Login with Google
    </button>
  </a>
</div>
`;
